### PR TITLE
Add neovim-lspconfig instructions

### PR DESCRIPTION
**Pull request by tnightingale**
### Motivation

Adding configuration instructions for Neovim (via neovim-lspconfig) as per [this comment](https://github.com/Shopify/ruby-lsp/issues/188#issuecomment-1456941648).

### Implementation

Just adding documentation.

### CODE REVIEW

File: EDITORS.md [hunk 1]
```diff
@@ -7,3 +7,22 @@ Code, use the official [Ruby LSP extension](https://github.com/Shopify/vscode-ru
 new H2 header in this file containing the instructions. -->
 
 - [Emacs LSP Mode](https://emacs-lsp.github.io/lsp-mode/page/lsp-ruby-lsp/)
+- [neovim](#neovim-via-nvim-lspconfig) (via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig))
+
+## Neovim (via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig))
+
+- Install [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) via your favourite plugin manager (eg: [Packer](https://github.com/wbthomason/packer.nvim)).
```
Comment (andyw8) [inline EDITORS.md:14]:
```suggestion
- Install [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) using your preferred plugin manager.
```
(I think we can trust people will already have one configured).

File: EDITORS.md [hunk 2]
```diff
@@ -7,3 +7,22 @@ Code, use the official [Ruby LSP extension](https://github.com/Shopify/vscode-ru
...
+- Enable nvim-lspconfig's ruby_ls configuration:
+  ```lua
+  require('lspconfig').ruby_ls.setup {
+    init_options = {
+      -- Add Ruby LSP configuration here, eg:
+      -- formatter = 'auto'
+    },
+    -- Add your lspconfig configurations/overrides here, eg:
```
Comment (andyw8) [inline EDITORS.md:22]:
```suggestion
    -- Add your lspconfig configurations/overrides here, e.g.:
```

**Review by andyw8 — CHANGES_REQUESTED** [review]

**Comment by andyw8** [comment]
I'm not familiar with neovim, so it would be helpful if we have could someone else from the [discussion](https://github.com/neovim/nvim-lspconfig/pull/2498) to verify this and add a comment here.

**Comment by vinistock** [comment]
Is dealing with `textDocument/diagnostic` the main source of disagreement in the discussions? If that's the main thing and this config works for everything else, I'd say let's move forward and people can put up PRs against the existing instructions to enhance it.

It's been a long time since the discussion started. At this point, it's better to have something than nothing in our instructions.

**Comment by andyw8** [comment]
Hi @tnightingale, I'll close this since Neovim instructions were merged in https://github.com/Shopify/ruby-lsp/pull/830. Feel free to open a new PR if there's anything you can think can be improved.

**Pull request closed without merging**